src/controller/CHIPDevice.cpp

```diff
@@ -622,6 +625,78 @@ void Device::OnSessionEstablished()
     }
 }
 
+void Device::ReleaseDAC()
+{
+    if (mDAC != nullptr)
+    {
+        Platform::MemoryFree(mDAC);
+    }
+    mDACLen = 0;
+    mDAC    = nullptr;
+}
+
+CHIP_ERROR Device::SetDAC(const ByteSpan & dac)
+{
+    if (dac.size() == 0)
+    {
+        ReleaseDAC();
+        return CHIP_NO_ERROR;
+    }
+
+    VerifyOrReturnError(dac.size() <= Crypto::kMax_x509_Certificate_Length, CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
There is also `static constexpr uint32_t kMaxDERCertLength  = 600;`. Curious why we use both `kMax_x509_Certificate_Length`

Reply:
Agree, we should consolidate to one of these constants.

Reply:
I guess to avoid circular dependency warnings in gn -
credentials have dependency on crypto.
Anyway, will update this to use the constant present in CHIPCert.h.